### PR TITLE
[PUPIL-780] Update OakPupilJourneyPreviousLessons so that text can be changed

### PR DESCRIPTION
**Pull request by BTitterington**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

This is an update so we can use OakPupilJourneyPreviousLessons within OWA, the text was previously hard coded to "Previously released lessons" now a react node can be passed in as the text.

In addition, I've renamed the component as its going to be used as the subheader on both legacy and new units. 

I've also renamed the counterSlot prop of the OakPupilJourneyList to subheadingSlot as this seems more appropriate now.

## A link to the component in the deployment preview

https://deploy-preview-235--lively-meringue-8ebd43.netlify.app/

### OakPupilJourneyList should still work.
https://deploy-preview-235--lively-meringue-8ebd43.netlify.app/?path=/story/components-organisms-pupil-oakpupiljourneylist--no-title 

### OakPupilJourneyListItemSubheading should still work, changing text will replace the textSlot which is initially the OakPupilJourneyListCounter 
https://deploy-preview-235--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-pupil-oakpupiljourneylistitemsubheading--docs

### CODE REVIEW

File: src/components/organisms/pupil/OakPupilJourneyListItemSubheading/OakPupilJourneyListItemSubheading.tsx [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
 
-import { OakPupilJourneyListCounter } from "../OakPupilJourneyListCounter";
-
 import { OakBulletList, OakBulletListProps } from "@/components/molecules";
 import { OakFlex } from "@/components/atoms";
 
-export type OakPupilJourneyPreviousLessonsProps = {
-  numberOfLessons: number;
+export type OakPupilJourneyListItemSubheadingProps = {
+  textSlot?: React.ReactNode;
```
Comment (kimon-satan) [inline src/components/organisms/pupil/OakPupilJourneyListItemSubheading/OakPupilJourneyListItemSubheading.tsx:7]: I'm a bit concerned that we might be "over-composing" our components (sorry can't think of a better term). Does this really need a dedicated component ? And if so does it really need to employ composition for the textSlot (ie. is it ever not an OakPupilJourneyListCounter )

File: src/components/organisms/pupil/OakPupilJourneyListItemSubheading/OakPupilJourneyListItemSubheading.tsx
Comment (kimon-satan) [inline src/components/organisms/pupil/OakPupilJourneyListItemSubheading/OakPupilJourneyListItemSubheading.tsx:1]: cheers for renaming this file

**Review by kimon-satan — APPROVED** [review]
No blockers but I'd like us to take stock of what we're doing in our components. Ideally before you move on Ben so that we can get your input.

**Comment by RBrownsett** [comment]
@BTitterington looks good - just checking when I edit the textSlot variable it loses its styling - is the formatting handled elsewhere? e.g:

image

**Review by RBrownsett — APPROVED** [review]
Approved - tested with no issues

**Pull request merged**